build2-packaging/sol2
# Release version 3.3.1+1

This pull request prepares the next revision.
It addresses Issue #1 and tweaks package and buildfiles according to the build2 packaging guide.

- Update `README.md`
- Apply workaround for bug in Clang 18 See: https://github.com/ThePhD/sol2/issues/1581
- Use symlinks for all library source files This allows to apply custom workarounds for external bugs.
- Run tests as independent package Currently, the latest `catch2` package from `cppget.org` fails to compile with Clang and leads to false-negative results.
- Add `sol2-examples` package with upstream examples
- Add `sol2-tests` package with upstream tests
- Update `manifest`
- Tweak and clean up `buildfile`s
- Add `README.md` and `PACKAGE-README.md` symlinks
- Add symlinks to Sol2 license file
- Transform into multi-package repository
- Add `.gitattributes`
- Update `.gitignore`
- Use standard build2 naming scheme None of the standard build2 filenames clash with upstream source code. Revert alternative naming scheme to be canonical.

All changes should be backwards compatible as required.
The current CI run can be found [here](https://ci.cppget.org/@eca35524-4e98-431b-aaf7-90bdffbcc420).
You can also read the section about issues and notes in `README.md`.
Hope you are ok with it.
